projects/igniteui-angular/src/lib/select/select-positioning-strategy.ts

```diff
@@ -1,231 +1,226 @@
-        } else {
-            return null;
+    /**
+     * Calculate selected item scroll position.
+     */
+    public calculateScrollPosition(itemElementRect: ClientRect, scrollContainer: HTMLElement): number {
+        if (!itemElementRect) {
+            return 0;
         }
-        return returnVals;
+
+        const scrollContainerRect = scrollContainer.getBoundingClientRect();
+        const scrollDelta = scrollContainerRect.top - itemElementRect.top;
+        let scrollPosition = scrollContainer.scrollTop - scrollDelta;
+
+        const dropDownHeight = scrollContainer.clientHeight;
+        scrollPosition -= dropDownHeight / 2;
+        scrollPosition += itemElementRect.height / 2;
+
+        return Math.floor(Math.min(Math.max(0, scrollPosition), scrollContainer.scrollHeight - scrollContainerRect.height));
     }
 
-    position(contentElement: HTMLElement, size: Size, document?: Document, initialCall?: boolean): void {
-        const inputElement = this.select.input.nativeElement;
-        const inputRect = inputElement.getBoundingClientRect() as DOMRect;
-        this.listContainerBoundRect = contentElement.getBoundingClientRect() as DOMRect;
-        const LIST_HEIGHT = this.listContainerBoundRect.height;
-        if (!initialCall) {
-            this.deltaX = inputRect.left - this.itemTextPadding - this.itemTextIndent;
-            const point = new Point(this.deltaX, inputRect.top + this.deltaY);
-            this.settings.target = point;
-            super.position(contentElement, size);
-            return;
+    /**
+     * Position the items outer container so selected item text is positioned over input text and if header
+     * And/OR footer - both header/footer are visible
+     * @param selectFit selectFit to use for computation.
+     */
+    protected fitInViewport(contentElement: HTMLElement, selectFit: SelectFit) {
+        const footer = selectFit.scrollContainer.getBoundingClientRect().bottom - selectFit.contentElementRect.bottom;
+        const lastItemFitSize = selectFit.targetRect.bottom + selectFit.styles.itemTextToInputTextDiff - footer;
+        // out of viewPort on Top
+        if (selectFit.fitVertical.back < 0) {
+            const possibleScrollAmount = selectFit.scrollContainer.scrollHeight -
+                selectFit.scrollContainer.getBoundingClientRect().height - selectFit.scrollAmount;
+            if (possibleScrollAmount + selectFit.fitVertical.back > 0) {
+                selectFit.scrollAmount -= selectFit.fitVertical.back;
+                selectFit.verticalOffset -= selectFit.fitVertical.back;
+                this.global_yOffset = selectFit.verticalOffset;
+            } else {
+                selectFit.verticalOffset = 0 ;
+                this.global_yOffset = 0;
+            }
+        // out of viewPort on Bottom
+        } else if (selectFit.fitVertical.forward < 0) {
+            if ( lastItemFitSize > selectFit.viewPortRect.bottom) {
```

Comment:
We have the same issue and at the top. What we need is if header/footer can fit above/bellow the input. This line is calculating the footer height, right:
```
const footer = selectFit.scrollContainer.getBoundingClientRect().bottom - selectFit.contentElementRect.bottom;
```
In similar way you may calculate the header height. Then you can check in the same if statement whether we have enough space to fit header/footer and if we have enough scroll to scroll the item like this (for the footer):
```
if (lastItemFitSize < selectFit.viewPortRect.bottom && selectFit.scrollAmount + selectFit.fitVertical.forward > 0) {
```
And this will make `positionAbove` unnecessary.

Reply:
done